File: src/main/java/cn/chahuyun/dockingGpt/entity/OpenAiMessage.java
package cn.chahuyun.dockingGpt.entity;

import cn.chahuyun.dockingGpt.constant.OpenAiMessageRoleEnum;
import lombok.Data;

@Data
public class OpenAiMessage {
    /**
     * 消息角色
     */
    private String role;
    /**
     * 用户名
     */
    private String name;
    /**
     * 内容
     */
    private String content;

    public OpenAiMessage(OpenAiMessageRoleEnum role, String content) {
        this.role = role.getValue();
        this.content = content;
    }

    public OpenAiMessage(OpenAiMessageRoleEnum role, String name, String content) {
        this.role = role.getValue();
        this.name = name;
        this.content = content;
    }

}

File: src/main/oldKotlin/cn/chahuyun/docking/entity/OpenAiMessage.kt
package cn.chahuyun.docking.entity

import cn.chahuyun.docking.RoleType
import lombok.Data

@Data
class Record {
    /**
     * 消息角色
     */
    private var role: String

    /**
     * 用户名
     */
    private var name: String? = null

    /**
     * 内容
     */
    var content: String

    constructor(role: RoleType, content: String) {
        this.role = role.value
        this.content = content
    }

    constructor(role: RoleType, name: String?, content: String) {
        this.role = role.value
        this.name = name
        this.content = content
    }
}
